v2/api/apimanagement/v1api20220801/api_version_set_types_gen.go

```diff
@@ -569,8 +570,13 @@ func (versionSet *Service_ApiVersionSet_Spec) AssignProperties_From_Service_ApiV
 
 	// VersioningScheme
 	if source.VersioningScheme != nil {
-		versioningScheme := ApiVersionSetContractProperties_VersioningScheme(*source.VersioningScheme)
-		versionSet.VersioningScheme = &versioningScheme
+		versioningScheme := *source.VersioningScheme
```

Comment:
I wrote this once before but GitHub ate it:

Consider instead of doing this in every generated method, creating a helper:
```
func CaseCorrectEnum[T ~string](val string, map[string]T) T
```

and then just call that in the generated code:
```
CaseCorrectEnum(versioningScheme, ApiVersionSetContractProperties_VersioningScheme_Cache)
```

Reply:
Good idea.

Reply:
https://go.dev/play/p/4QlmlZearvY?v=goprev

Reply:
Done.